stan/math/fwd/mat/vectorize/apply_scalar_binary.hpp

```diff
@@ -0,0 +1,151 @@
+       *
+       * @param x Argument fvar<FV>iable.
+       * @param y Int argument
+       * @return Function applied to the fvar<FV>iable.
+       */
+      static inline return_t apply(const stan::math::fvar<FV>& x, int y) {
+        return F::fun(x, static_cast<double>(y));
+      }
+    };
+
+    /**
+     * Template specialization for vectorized functions applying to
+     * fvar<FV> and fvar<FV> arguments.
+     *
+     * @tparam F Type of function defining static apply function.
+     * @tparam FV Value and tangent type for for forward-mode
+     * autodiff variable.
+     */
+    template <typename F, typename FV>
+    struct apply_scalar_binary<F, int, stan::math::fvar<FV> > {
+      /**
+       * Function return type, which is <code>fvar<FV></code>.
+       */
+      typedef stan::math::fvar<FV> return_t;
+
+      /**
+       * Apply the function specified by F to the specified arguments.
+       *
+       * @param x Int argument.
+       * @param y Argument fvar<FV>iable.
+       * @return Function applied to the fvar<FV>iable.
+       */
+      static inline return_t apply(int x, const stan::math::fvar<FV>& y) {
+        return F::fun(static_cast<double>(x), y);
+      }
+    };
+
+    /**
+     * Template specialization for vectorized functions applying to
+     * fvar<FV> and fvar<FV> arguments.
+     *
+     * @tparam F Type of function defining static apply function.
+     * @tparam FV Value and tangent type for for forward-mode
+     * autodiff variable.
+     */
+    template <typename F, typename FV>
+    struct apply_scalar_binary<F, stan::math::fvar<FV>, double> {
+      /**
+       * Function return type, which is <code>fvar<FV></code>.
+       */
+      typedef stan::math::fvar<FV> return_t;
+
+      /**
+       * Apply the function specified by F to the specified arguments.
+       *
+       * @param x Argument fvar<FV>iable.
+       * @param y Double argument.
+       * @return Function applied to the fvar<FV>iable.
+       */
+      static inline return_t apply(const stan::math::fvar<FV>& x,
+      double y) {
+        return F::fun(x, y);
+      }
+    };
+
+    /**
+     * Template specialization for vectorized functions applying to
+     * fvar<FV> and fvar<FV> arguments.
+     *
+     * @tparam F Type of function defining static apply function.
+     * @tparam FV Value and tangent type for for forward-mode
+     * autodiff variable.
+     */
+    template <typename F, typename FV>
+    struct apply_scalar_binary<F, double, stan::math::fvar<FV> > {
+      /**
+       * Function return type, which is <code>fvar<FV></code>.
+       */
+      typedef stan::math::fvar<FV> return_t;
+
+      /**
+       * Apply the function specified by F to the specified arguments.
+       *
+       * @param x Double argument.
+       * @param y Argument fvar<FV>iable.
+       * @return Function applied to the fvar<FV>iable.
+       */
+      static inline return_t apply(double x, const
+      stan::math::fvar<FV>& y) {
```

Comment:
spacing---last time I'll say it, but it needs to be fixed everywhere.  Put everything on one line when you can.